pkg/webui/console/views/application-datavis/index.js

```diff
@@ -51,22 +51,34 @@ const ApplicationDataVisualization = () => {
   const [availableDevices, setAvailableDevices] = useState({})
   const [loading, setLoading] = useState(true)
 
-  const [selectedSensor, setSelectedSensor] = useState('')
+  const [selectedColumns, setSelectedColumns] = useState('')
   const [selectedTime, setSelectedTime] = useState('1H')
   const [graphData, setGraphData] = useState([])
-  const availableReadingColumns = {
+  const [availableReadingColumns, setAvailableReadingColumns] = useState({})
+  const deviceToAvailableColumns = {
+    Temperature: ['Temperature'],
+    'Soil Moisture': ['Soil Moisture', 'Soil Conductivity', 'Soil Temperature'],
+    'Temperature D20': ['Temperature Red'],
+    'Temperature D22': ['Temperature Red', 'Temperature White'],
+    'Temperature D23': ['Temperature Red', 'Temperature White', 'Temperature Black'],
+  }
+  const columnToKeyMap = {
     Temperature: 'temperature',
     'Soil Moisture': 'water_SOIL',
     'Soil Conductivity': 'conduct_SOIL',
     'Soil Temperature': 'temp_SOIL',
+    'Temperature Red': 'Temp_Red',
+    'Temperature White': 'Temp_White',
+    'Temperature Black': 'Temp_Black',
   }
 
-
-  
   const sensorLabels = {
     water_SOIL: 'Soil Moisture',
     conduct_SOIL: 'Soil Conductivity',
     temp_SOIL: 'Soil Temperature',
+    Temp_Red: 'Temperature Red',
+    Temp_White: 'Temeprature White',
```

Comment:
we should move the sensor labels to the backend and unify the logic around them. So then we can pass up 'air_temperature' .... then maybe in the metadata include the original fields and the mapping back to them